x/observer/keeper/msg_server_disable_fast_confirmation.go

```diff
@@ -0,0 +1,62 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	cosmoserror "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	authoritytypes "github.com/zeta-chain/node/x/authority/types"
+	"github.com/zeta-chain/node/x/observer/types"
+)
+
+// DisableFastConfirmation disables fast confirmation for the given chain ID
+// Inbound and outbound will be only confirmed using SAFE confirmation count on disabled chains
+func (k msgServer) DisableFastConfirmation(
+	goCtx context.Context,
+	msg *types.MsgDisableFastConfirmation,
+) (*types.MsgDisableFastConfirmationResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check permission
+	err := k.GetAuthorityKeeper().CheckAuthorization(ctx, msg)
+	if err != nil {
+		return nil, cosmoserror.Wrap(authoritytypes.ErrUnauthorized, err.Error())
+	}
+
+	// find current chain params list
+	chainParamsList, found := k.GetChainParamsList(ctx)
+	if !found {
+		return nil, types.ErrChainParamsNotFound
+	}
+
+	// disable fast confirmation
+	foundChain := false
+	for _, cp := range chainParamsList.ChainParams {
+		if cp.ChainId != msg.ChainId {
+			continue
+		}
+
+		if cp.ConfirmationParams == nil {
+			return nil, types.ErrInvalidChainParams
+		}
+
+		// setting fast confirmation count to same value as safe confirmation count
+		// will effectively disable fast confirmation
+		foundChain = true
+		cp.ConfirmationParams.FastInboundCount = cp.ConfirmationParams.SafeInboundCount
+		cp.ConfirmationParams.FastOutboundCount = cp.ConfirmationParams.SafeOutboundCount
```

Comment:
```suggestion
		if cp.ChainId == msg.ChainId {
			if cp.ConfirmationParams == nil {
				return nil, types.ErrInvalidChainParams
			}

			// setting fast confirmation count to same value as safe confirmation count
			// will effectively disable fast confirmation
			foundChain = true
			cp.ConfirmationParams.FastInboundCount = cp.ConfirmationParams.SafeInboundCount
			cp.ConfirmationParams.FastOutboundCount = cp.ConfirmationParams.SafeOutboundCount
			break
		}	
```

I think it's cleaner this way

Reply:
do you really want the additional break? I didn't add in case we have duplicate chain ids.

Reply:
updated in commit https://github.com/zeta-chain/node/pull/3522/commits/24d6b53ab91c6f5ef082b3718d14b5076d4a330c